app/models/merchant.rb

```diff
@@ -39,4 +39,15 @@ def self.top_five_merchants
       ORDER BY sum(invoice_items.unit_price * invoice_items.quantity) DESC
       LIMIT 5")
   end
-end
+
+  def top_sales_day  
+    self.invoices.joins(:transactions, :invoice_items)
+      .where("transactions.result = 0")
+      .select("invoices.created_at, sum(invoice_items.quantity * invoice_items.unit_price) AS invoice_revenue")
```

Comment:
Do we need to have successful transactions? If so, this method won't work, you are duplicating data by joining to the invoice_items table twice

Reply:
As discussed in our stand up this issue has been resloved!